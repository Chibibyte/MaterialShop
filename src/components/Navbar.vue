<template>
  <v-container class="pa-0 ma-0">
    <v-app-bar app class="pa-0 ma-0" color="primaryDarkened" dark>
      <v-container class="d-flex justify-space-between pa-0 ma-0">
        <NavButton
          routerPush
          to="/"
          icon="mdi-shopping-search"
          name="Shopping"
        />
        <NavButton
          routerPush
          to="/cart"
          icon="mdi-cart"
          :name="'Cart (' + amount + ')'"
        />
        <CheckoutButton />
        <v-btn fab small @click="infoPopup">
          <v-icon small>mdi-help</v-icon>
        </v-btn>
      </v-container>
    </v-app-bar>

    <!-- INFO MODAL -->
    <v-dialog v-model="dialog" max-width="290">
      <v-card>
        <v-card-title class="headline">Material Shop</v-card-title>
        <v-card-text class="orange--text"
          >!Disclaimer! This is not a real shop!</v-card-text
        >
        <v-card-text>
          Welcome to the Material Shop
          <br />Pretend to buy the latest Material Design Icons <br />(or get
          them for free at
          <br />
          <a href="https://materialdesignicons.com"
            >https://materialdesignicons.com</a
          >)
        </v-card-text>
        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn color="green darken-1" text @click="dialog = false"
            >Close</v-btn
          >
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-container>
</template>

<script>
import NavButton from "@/components/NavButton.vue";
import CheckoutButton from "@/components/CheckoutButton.vue";
export default {
  name: "Navbar",
  components: {
    NavButton,
    CheckoutButton,
  },
  data() {
    return {
      dialog: false,
    };
  },
  computed: {
    amount() {
      return this.$store.getters["noncheckout/amount"];
    },
  },
  methods: {
    infoPopup() {
      this.dialog = true;
    },
  },
};
</script>