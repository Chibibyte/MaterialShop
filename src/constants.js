export default {
    BUTTONICONS: {
        AUTHENTICATION: 'mdi-login-variant',
        ADDRESS: 'mdi-package-variant',
        BANKACCOUNT: 'mdi-bank',
        SHIPPING: 'mdi-truck',
        PAYMENT: 'mdi-cash',
        SUMMARY: 'mdi-text-box-check-outline',
        CHECKOUT_COMPLETE: 'mdi-check',
        ADD: 'mdi-plus',
        REMOVE: 'mdi-minus',
        CANCEL: 'mdi-window-close',
        CONFIRM: 'mdi-check',
        SHOPPING: 'mdi-shopping',
        SUCCESS: 'mdi-check',
        FAILURE: 'mdi-window-close',
        WARNING: 'mdi-alert-circle-outline'
    }
}