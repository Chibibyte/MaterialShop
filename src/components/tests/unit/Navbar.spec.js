
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Navbar from '@/components/Navbar.vue';
import constants from '@/constants.js'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.prototype.$constants = constants;

let store;
let state;
let modules;
let mutations;
let mocks;
let stubs = [
    'v-container',
    'v-app-bar',
    'v-icon',
    'v-btn',
    'v-dialog',
    'v-card',
    'v-card-title',
    'v-card-text',
    'v-card-actions',
    'v-spacer',
]
let wrapper;

beforeEach(() => {
    mutations = {}
    state = {}
    modules = {
        checkout: {
            namespaced: true,
            state() { },
            mutations: {
                setOrder: jest.fn()
            }
        },
        noncheckout: {
            namespaced: true,
            state() {
                return {
                    cart: []
                }
            },
            getters: {
                amount: jest.fn()
            }
        }
    }

    mocks = {
        $router: { push: jest.fn() }
    };
    store = new Vuex.Store({
        mutations,
        state,
        modules
    })
    wrapper = shallowMount(Navbar, { mocks, stubs, store, localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })

    test('state.dialog: default', () => {
        expect(wrapper.vm.dialog).toBeFalsy();
    });
});

// computed
describe('computed: amount', () => {
    test('default', () => {
        expect(modules.noncheckout.getters.amount.mock.calls).toHaveLength(1);
    })
})

// method
describe('method: infoPopup', () => {
    test('default', () => {
        wrapper.vm.infoPopup();
        expect(wrapper.vm.dialog).toBeTruthy();
    })
})