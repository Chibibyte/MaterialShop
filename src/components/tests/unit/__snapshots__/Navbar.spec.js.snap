// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<v-container-stub
  class="pa-0 ma-0"
>
  <v-app-bar-stub
    app=""
    class="pa-0 ma-0"
    color="primaryDarkened"
    dark=""
  >
    <v-container-stub
      class="d-flex justify-space-between pa-0 ma-0"
    >
      <navbutton-stub
        icon="mdi-shopping-search"
        name="Shopping"
        routerpush="true"
        to="/"
      />
       
      <navbutton-stub
        icon="mdi-cart"
        name="Cart (undefined)"
        routerpush="true"
        to="/cart"
      />
       
      <navbutton-stub
        disabled="true"
        icon="mdi-cash-register"
        name="Checkout"
        routerpush="true"
        runbefore="function () { return this$1.$store.commit('checkout/setOrder', [].concat( this$1.$store.state.noncheckout.cart )); }"
        to="/checkout"
      />
       
      <v-btn-stub
        fab=""
        small=""
      >
        <v-icon-stub
          small=""
        >
          mdi-help
        </v-icon-stub>
      </v-btn-stub>
    </v-container-stub>
  </v-app-bar-stub>
   
  <v-dialog-stub
    max-width="290"
  >
    <v-card-stub>
      <v-card-title-stub
        class="headline"
      >
        Material Shop
      </v-card-title-stub>
       
      <v-card-text-stub
        class="orange--text"
      >
        !Disclaimer! This is not a real shop!
      </v-card-text-stub>
       
      <v-card-text-stub>
        
        Welcome to the Material Shop
        
        <br />
        Pretend to buy the latest Material Design Icons
        
        <br />
        (or get them for free at
        
        <br />
         
        <a
          href="https://materialdesignicons.com"
        >
          https://materialdesignicons.com
        </a>
        )
      
      </v-card-text-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          color="green darken-1"
          text=""
        >
          Close
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</v-container-stub>
`;
